src/graphics/renderable.js

```diff
@@ -80,6 +79,40 @@ Crafty.c("Renderable", {
         this._defineRenderableProperites();
     },
 
+    // Renderable assumes that a draw layer has 3 important methods: attach, detach, and dirty
+
+    // Dirty the entity when it's invalidated
+    _invalidateRenderable: function() {
+        //flag if changed
+        if (this._changed === false) {
+                this._changed = true;
+            this._drawLayer.dirty(this);
+        }
+    },
+
+    // Attach the entity to a layer to be rendered
+    _attachToLayer: function(layer) {
+        if (this._drawLayer) {
+            this._detachFromLayer();
+        }
+        this._drawLayer = layer;
+        layer.attach(this);
+        this.bind("Invalidate", this._invalidateRenderable);
+        this.trigger("LayerAttached", layer);
+        this.trigger("Invalidate");
```

Comment:
Can we change this, so that the entity gets dirtied when added to the layer, instead of triggering `Invalidate`? (to be consistent how the entity gets dirtied when detached from layer)